sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,252 @@
+#title          :discretization.py
+#description    :Implementation of a few discretization algorithms
+#author         :Henry Lin <hlin117@gmail.com>
+#version        :0.0.1
+#python_version :2.7.6
+#==============================================================================
+
+from collections import Counter
+from math import log
+import numpy as np
+from itertools import izip, islice
+
+
+def log2(x):
+    return log(x, 2) if x > 0 else 0
+
+
+class MDLP(object):
+    """Implements the MDLP algorithm from Usama Fayyad's paper
+    "Multi-Interval Discretization of Continuous-Valued Attributes for
+    Classification Learning"
+    """
+
+    def __init__(self):
+        self.intervals = dict()
```

Comment:
This should be named `intervals_` as with other output attributes in the project. Please also document input parameters and output attributes appropriately in the class docstring.